### DISCUSSION TITLE
Rpi pico

### DISCUSSION THREAD

**Discussion by BadPotato1007** [shorepine/tulipcc#288] [General]
Any chance of getting support for rpi pico? It's a lot more powerful for this stuff! Pico w would still have the wifi functionality!

**Comment by bwhitman** [comment]
The underlying syntheszier, [AMY](https://github.com/shorepine/amy) does support the Pi Pico quite well.

I don't see a lot of evidence for the Pi Pico/W being "more powerful", as far as I know it cannot support the display resolutions we have on Tulip, and we see higher synth polyphony (but it's barely a tie) on the ESP32S3 running at the same clock speed. The S3's GDMA channels for the hi-res display, on-chip PSRAM at 120MHz and 2nd core with no stack limits make it much more applicable for a thing like Tulip. But we're certainly not tied to it, and if there's a RP2040 successor that gets closer, we'll take a look.

**Reply by infovore** [reply]
Co-sign; I'm pretty sure a dual-core Cortex-M0+ is definitely not more powerful, in the context given.

**Reply by coolcoder613eb** [reply]
The Pi Pico 2 *has* been released.

**Reply by bwhitman** [reply]
It has! And it has a useful HSTX peripheral for displays. But I haven't yet seen anyone take it to the limits that Tulip currently has with the ESP32S3. If you see someone doing 1024x600 or higher resolutions on a RP2530, let me know!

**Comment by znmeb** [comment]
Indeed - the Pico W isn't powerful enough. However, the $15US Raspberry Pi Zero 2 W should run Tulip just fine, either in Linux (32 or 64 bits) or bare metal. I have a few of them and can test it if anyone's interested.
